ipn/ipnlocal/local.go

```diff
@@ -1969,16 +1964,23 @@ func (b *LocalBackend) setPrefsLockedOnEntry(caller string, newp *ipn.Prefs) {
 		cc.Login(nil, controlclient.LoginDefault)
 	}
 
+	if oldp.RunSSH && oldp.WantRunning {
+		if !newp.WantRunning || !newp.RunSSH {
+			b.mu.Lock()
```

Comment:
why re-acquiring the lock? can this be done before the unlock on 1930?

Reply:
Good catch, done.